server/src/main/java/com/linecorp/centraldogma/server/internal/admin/auth/FileBasedSessionManager.java

```diff
@@ -0,0 +1,324 @@
+        return sessionId2Path(rootDir, sessionId);
+    }
+
+    private static Path sessionId2Path(Path rootDir, String sessionId) {
+        checkArgument(SESSION_ID_PATTERN.matcher(sessionId).matches(),
+                      "sessionId: %s (expected: UUID)", sessionId);
+        return rootDir.resolve(sessionId.substring(0, SESSION_ID_1ST_PART_LENGTH))
+                      .resolve(sessionId.substring(SESSION_ID_1ST_PART_LENGTH));
+    }
+
+    private static boolean isSessionFile(Path path) {
+        if (!Files.isRegularFile(path)) {
+            return false;
+        }
+        final int nameCount = path.getNameCount();
+        if (nameCount < 2) {
+            return false;
+        }
+
+        final String first = path.getName(nameCount - 2).toString();
+        if (!SESSION_ID_1ST_PART_PATTERN.matcher(first).matches()) {
+            return false;
+        }
+
+        final String second = path.getName(nameCount - 1).toString();
+        return SESSION_ID_2ND_PART_PATTERN.matcher(second).matches();
+    }
+
+    /**
+     * Serializes a {@link Session} into a byte array using an {@link ObjectOutputStream}.
+     */
+    static byte[] serialize(Session session) throws IOException {
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream(1024);
+             ObjectOutputStream oos = new ObjectOutputStream(baos)) {
+            oos.writeObject(session);
+            oos.flush();
+            return baos.toByteArray();
+        }
+    }
+
+    /**
+     * Deserializes a {@link Session} from a byte array using an {@link ObjectInputStream}.
+     */
+    static Session deserialize(byte[] encoded) throws IOException, ClassNotFoundException {
+        try (ByteArrayInputStream bais = new ByteArrayInputStream(encoded);
+             ObjectInputStream ois = new ObjectInputStream(bais)) {
+            return (Session) ois.readObject();
+        }
+    }
+
+    /**
+     * A job for deleting expired sessions from the file system.
+     */
+    public static class ExpiredSessionDeletingJob implements Job {
+        @Override
+        public void execute(JobExecutionContext context) throws JobExecutionException {
+            try {
+                logger.debug("Started {} job.", ExpiredSessionDeletingJob.class.getSimpleName());
+                final Path rootDir = (Path) context.getJobDetail().getJobDataMap().get(ROOT_DIR);
+                final Instant now = Instant.now();
+                Files.walk(rootDir, 2)
+                     .filter(FileBasedSessionManager::isSessionFile)
+                     .map(path -> {
+                         try {
+                             return deserialize(Files.readAllBytes(path));
+                         } catch (FileNotFoundException | NoSuchFileException ignored) {
+                             // Session deleted by other party.
+                         } catch (Exception e) {
+                             logger.warn("Failed to deserialize a session: {}", path, e);
```

Comment:
Perhaps we should delete a bad session file so we do not meet this again in the next run?